crates/wit-component/src/validation.rs

```diff
@@ -673,6 +684,20 @@ fn validate_func(
     )
 }
 
+fn validate_post_return(
+    resolve: &Resolve,
+    ty: &wasmparser::FuncType,
+    func: &Function,
+) -> Result<()> {
+    let mut sig = resolve.wasm_signature(AbiVariant::GuestExport, func);
+    sig.params = mem::take(&mut sig.results);
```

Comment:
A comment here might be helpful -- took me a minute to realize this was because the parameters to a post-return function should match the results of the primary function.